bids/analysis/transformations/compute.py

```diff
@@ -33,8 +33,13 @@ def _transform(self, var, model='spm', derivative=False, dispersion=False,
 
         model = model.lower()
 
+        post_resample = False
         if isinstance(var, SparseRunVariable):
             sr = self.collection.sampling_rate
+            min_sampling_rate = 1.0 / var.duration.min()
```

Comment:
I think this is suboptimal solution since it potentially makes sampling rate way too high for no obvious benefit (keeping in mind dealing with slow fMRI) besides more streamlined implementation, but at possibly huge cost of computation time -- @AdinaWagner has events with durations as short as 10ms, and needs to wait for an hour just to get designs.

What about
- set sampling rate to `sr / oversampling_ratio` (where `oversampling_ratio`  is e.g. 2 or 4)
- while "treating" an event of duration Tevent and amplitude A, for a time bin Tbin (of duration dTbin, e.g. 500ms) for that even assign proportion of its amplitude residing in that time bin. E.g. if entire duration Tevent falls within the bin Tbin, its amplitude would become A*Tevent/dTbin.  For events which go across multiple bins, it would be a corresponding proportion of the event in that Tbin relative to overall Tevent.

note: this would work for linear design transformation (e.g. convolution) but would not be appropriate if it is something else, but since we are talking about linear models etc, should be Ok, but might still be some kind of option to decide on how to "treat" the events.

What do you think?

Reply:
That's a possibility. I'm afraid that mixing amplitude and duration is going to be a mess, but perhaps it would be more principled.

As an alternative to consider, we could aim to maintain the relationship: `oversampling * sr ~ 50 * self.collection.sampling_rate`

```Python
if min_sampling_rate > sr:
    oversampling = max(1, int(50.0 * sr / min_sampling_rate))
    sr = min_sampling_rate
```

Reply:
I'm not sure this is something that we need to try to solve algorithmically... @yarikoptic, in your case, with events of 10 ms, what happens if you downsample to a more reasonable rate before convolution? In practice, won't the interpolation just do roughly what you're proposing (i.e., assign weight proportional to its amplitude to each bin)? Have you investigated the empirical impact of early downsampling on the design matrix? Unless the information loss is pretty dramatic, it seems reasonable to me to say that this falls on the user: if you really think you need to maintain fine-grained resolution, you can wait an hour for your design matrices. If you're in a rush, or don't care that much, you downsample first, and then get them back much sooner.

I guess I'm worried that trying to do a bunch of behind-the-scenes optimization will run us into weird edge cases and make the code harder to maintain. But if you think there's a simple solution to this, then by all means, optimize away... :)